### PR TITLE
Make Logs Type Non-internal

### PR DESCRIPTION
**Pull request by keitwb**
This allows Logs to be used in the contrib repo.

Most of the references were fixed by sed + 'make fmt'.

### CODE REVIEW

File: consumer/pdata/generated_log_test.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func TestResourceLogsSlice_Append(t *testing.T) {
 	es.Append(&emptyVal)
 	assert.EqualValues(t, *(es.At(7)).orig, *emptyVal.orig)
 
-	emptyVal2:= NewResourceLogs()
```
Comment (tigrannajaryan) [inline consumer/pdata/generated_log_test.go:137]: You probably don't want to touch generated files.

File: internal/data/opentelemetry-proto-gen/collector/logs/v1/logs_service.pb.go [hunk 2]
```diff
@@ -6,14 +6,16 @@ package v1
 import (
 	context "context"
 	fmt "fmt"
+	io "io"
```
Comment (tigrannajaryan) [inline internal/data/opentelemetry-proto-gen/collector/logs/v1/logs_service.pb.go:9]: Why is this file changed? Did you regenerate the protobufs?

File: consumer/pdata/log.go [hunk 3]
```diff
@@ -14,14 +14,73 @@
...
+// This is a reference type (like builtin map).
+//
+// Must use NewLogs functions to create new instances.
+// Important: zero-initialized instance is not valid for use.
+type Logs struct {
+	orig *[]*otlplogs.ResourceLogs
+}
+
+// LogsFromProto creates the internal Logs representation from the ProtoBuf.
+func LogsFromProto(orig []*otlplogs.ResourceLogs) Logs {
```
Comment (tigrannajaryan) [inline consumer/pdata/log.go:36]: For naming consistency can we rename this to `LogsFromOtlp`? (We use `TracesFromOtlp` for traces).

File: consumer/pdata/log.go [hunk 4]
```diff
@@ -14,14 +14,73 @@
...
+	return Logs{&orig}
+}
+
+// LogsToProto converts the internal Logs to the ProtoBuf.
+func LogsToProto(ld Logs) []*otlplogs.ResourceLogs {
```
Comment (tigrannajaryan) [inline consumer/pdata/log.go:41]: Similarly `LogsToOtlp`.

File: consumer/pdata/generated_log_test.go (hunk 1 above)
Comment (keitwb) [reply consumer/pdata/generated_log_test.go:137]: The `make fmt` command changed it so I left it.  Those kind of make targets ideally should be noops on the master branch so this will make it so.

Comment (keitwb) [reply consumer/pdata/generated_log_test.go:137]: Also CircleCI should be flagging this kind of stuff too.

File: consumer/pdata/log.go (hunk 3 above)
Comment (keitwb) [reply consumer/pdata/log.go]: Done.

File: consumer/pdata/log.go (hunk 4 above)
Comment (keitwb) [reply consumer/pdata/log.go]: Done.

File: internal/data/opentelemetry-proto-gen/collector/logs/v1/logs_service.pb.go (hunk 2 above)
Comment (keitwb) [reply internal/data/opentelemetry-proto-gen/collector/logs/v1/logs_service.pb.go:9]: `goimports` changed it since standard libraries should go in the first group of imports.

File: consumer/pdata/generated_log_test.go (hunk 1 above)
Comment (tigrannajaryan) [reply consumer/pdata/generated_log_test.go:137]: I think we should generally avoid touching generated files otherwise when regenerating it will be difficult to tell what is the result of regeneration and what is the result of postgeneration fix.

File: internal/data/opentelemetry-proto-gen/collector/logs/v1/logs_service.pb.go (hunk 2 above)
Comment (tigrannajaryan) [reply internal/data/opentelemetry-proto-gen/collector/logs/v1/logs_service.pb.go:9]: I think we need to prohibit goimport from touching generated files. I haven't seen happening the past, not sure why goimport is doing it now.

File: consumer/pdata/generated_log_test.go (hunk 1 above)
Comment (keitwb) [reply consumer/pdata/generated_log_test.go:137]: @tigrannajaryan See https://github.com/open-telemetry/opentelemetry-collector/pull/1458.

**Comment by keitwb** [comment]
Ok, rebased on master and the protobuf stuff is no longer part of this PR.
